### PR TITLE
Wrap `parse_url` calls in `capture_internal_exceptions`

### PR DESCRIPTION
**Pull request by sentrivana**
Follow up to https://github.com/getsentry/sentry-python/issues/2160

Our URL parsing and sanitizing can fail. If it does, the error must never break the user's application.

This PR:

- wraps all calls to `parse_url` in `capture_internal_exceptions` and skips setting any parsed URL related data on the span (affects the `stdlib`, `boto3`, `httpx` integrations)
- adds more test cases for the `sanitize_url` function with `split=True`

### CODE REVIEW

File: sentry_sdk/integrations/httpx.py
Comment (sentrivana) [inline sentry_sdk/integrations/httpx.py:52]: Better placeholder name ideas very welcome

**Review by antonpirker — APPROVED** [review]
Fine to merge if we use `SENSITIVE_DATA_SUBSTITUTE` as placeholder. But it is up to you, if you think thats a stupid idea also fine to merge right away.

File: sentry_sdk/integrations/httpx.py
Comment (sentrivana) [reply sentry_sdk/integrations/httpx.py]: Changed. :)

**Pull request merged**
